components/authz-service/storage/v2/memstore/memstore.go

```diff
@@ -281,8 +281,14 @@ func (s *State) ListRules(_ context.Context) ([]*storage.Rule, error) {
 	return rules, nil
 }
 
-func (s *State) DeleteRule(_ context.Context, id string) error {
-	// TODO implement
+func (s *State) DeleteRule(ctx context.Context, id string) error {
+	_, err := s.GetRule(ctx, id)
```

Comment:
Does this need a NotFound case?

Reply:
the GetRule's NotFound error should bubble up